### ISSUE TITLE
Allow comments at top of Dockerfile when using Dockerfile task

### ISSUE THREAD

**Issue by denisharder** [bmuschko/gradle-docker-plugin#657]
### Expected Behavior
When using the DockerFile task and importing from a template, it should ignore any first lines that are comments when looking for first FROM (or ARG) statement. When I build a docker image using default docker behavior, this does not cause an issue.

### Current Behavior
Currently, if there are comments at the top of the DockerFile, it errors with message expecting FROM (or ARG) at top of Dockerfile. 

### Context
I  have many existing Dockerfiles within my source code that contain comments at top of file. I would like to keep them, and also would not like to write a gradle task that strips them out (although I could).

### Steps to Reproduce (for bugs)
1. Create a dockerfile with a comment at very top, followed by FROM.
2. Use DockerFile task instructionsFromTemplate passing in dockerfile location.

**Comment by denisharder** [comment]
After playing around with this.. it's a relatively simple fix, but it would still be nice to have.

If dockerfile looks like:
```
# A possible comment regarding the OS we are using.
FROM alpine:latest

MAINTAINER DevOps <devops@company.com>

# Another comment about the app we are installing.
COPY app.jar /apps/service/app.jar
```

What I would like todo:
```
project.tasks.create(name: 'copyDockerfile', type: Dockerfile) {
    // Add template dockerfile instructions.
    instructionsFromTemplate(project.dockerFile)
    // Add custom instructions.
    environmentVariable('MY_ENV_VAR', 'MY_ENV_VALUE')
}
```

What I ended up doing:
```
project.tasks.create(name: 'copyDockerfile', type: Dockerfile) {
    // Add template dockerfile instructions (excluding comments).
    project.file(project.dockerFile).each { line ->
            if(!line.startsWith('#')) {
                    instruction(line)
            }
    }
    // Add custom instructions.
    environmentVariable('MY_ENV_VAR', 'MY_ENV_VALUE')
}
```

**Comment by cdancy** [comment]
Yeah this is weird. Would you be interested in submitting a PR for this fix/work?

**Comment by bmuschko** [comment]
@denisharder Just wondering...why do you have a comment in the template if it isn't a valid Dockerfile in the first place? I don't think we should treat it as something special. Apart from that the templating functionality is something that was added to this plugin but isn't really a functionality in plain Docker. I see that you already found a workaround in your own code. I suggest you just use that so we can keep the functionality of the plugin simpler.

**Comment by denisharder** [comment]
@bmuschko … perhaps I am using your functionality not as intended, but the “template” is actually a valid dockerfile. At my company, we have 100s of repos with fully functioning dockerfiles residing in the source code. On a local developer machine, they can build a docker image with gradle or using basic docker commands. From my perspective, we build the repos thru our CICD pipeline and inject additional “devops” instructions (env vars, etc.) into the docker image for deployment (where your gradle plugin is extremely helpful). As such, I am actually not the owner of the dockerfiles and/or possible comments.

The above is to give some context, but after diving into your code the solution presented itself. The enhancement is definitely not essential. Thanks for the gradle plugin, it is very helpful.

**Comment by bmuschko** [comment]
I had a look at this again and saw that the [validation logic](https://github.com/bmuschko/gradle-docker-plugin/blob/master/src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy#L45-L55) is part of the `Dockerfile` task type implementation. At the moment it doesn't take into account comments as one of the first entries in the Dockerfile so we'll need to enhance the logic.

**Comment by bmuschko** [comment]
As a side note: This issue is not specific to using Dockerfile templates. It happens with any Dockerfile that contains comments before a `FROM` or `ARG` instruction.
